pkg/probe/probe.go

```diff
@@ -117,6 +124,93 @@ func (p *prober) probe(ctx context.Context, signals chan Signal) error {
 	return nil
 }
 
+func (p *prober) sendProxyProtocolV2Headers(ctx context.Context,
+	localIp, remoteIp, localPortStr, remotePortStr string,
+) error {
+	log := util.CtxLogOrPanic(ctx)
+
+	localPort, err := strconv.Atoi(localPortStr)
+	if err != nil {
+		return fmt.Errorf("failed to convert local port to int: %w", err)
+	}
+	remotePort, err := strconv.Atoi(remotePortStr)
+	if err != nil {
+		return fmt.Errorf("failed to convert remote port to int: %w", err)
+	}
+
+	header := &proxyproto.Header{
+		Version:           2,
+		Command:           proxyproto.PROXY,
+		TransportProtocol: proxyproto.TCPv4,
+		SourceAddr: &net.TCPAddr{
+			IP:   net.ParseIP(localIp),
+			Port: localPort,
+		},
+		DestinationAddr: &net.TCPAddr{
+			IP:   net.ParseIP(remoteIp),
+			Port: remotePort,
+		},
+	}
+
+	log.Info("sending proxy protocol v2 headers", zap.Any("header", header))
+	_, err = header.WriteTo(p.conn)
+	if err != nil {
+		return fmt.Errorf("failed to write proxy protocol v2 headers: %w", err)
+	}
+	fmt.Printf("PROXY protocol v2 headers sent.\n")
+	return nil
+}
+
+func (p *prober) maybeSendProxyProtocolHeaders(ctx context.Context, signals chan Signal) {
+	log := util.CtxLogOrPanic(ctx)
+
+	if p.sendProxyProtocol == "" {
+		return
+	}
+
+	if p.conn == nil {
+		log.Info("no tcp connection, skipping proxy protocol headers")
+		return
+	}
+
+	local := p.conn.LocalAddr().String()
+	localIp, localPort, err := net.SplitHostPort(local)
+	if err != nil {
+		signals <- Signal{Path: "PROXYPROTOCOL/ERROR", Error: fmt.Errorf("failed to split local address: %w", err)}
+		return
+	}
+
+	remote := p.conn.RemoteAddr().String()
+	remoteIp, remotePort, err := net.SplitHostPort(remote)
+	if err != nil {
+		signals <- Signal{Path: "PROXYPROTOCOL/ERROR", Error: fmt.Errorf("failed to split remote address: %w", err)}
+		return
+	}
+
+	if p.sendProxyProtocol == "v2" {
+		if err := p.sendProxyProtocolV2Headers(ctx, localIp, remoteIp, localPort, remotePort); err != nil {
+			signals <- Signal{Path: "PROXYPROTOCOL/V2/ERROR", Error: err}
+		} else {
+			signals <- Signal{Path: "PROXYPROTOCOL/V2/SENT", Message: fmt.Sprintf("local: %s:%s, remote: %s:%s", localIp, localPort, remoteIp, remotePort)}
+		}
+		return
+	}
+
+	ppHeader := fmt.Sprintf("PROXY TCP4 %s %s %s %s\r\n",
```

Comment:
I would expect another `sendProxyProtocolV1Headers` method fr this, it would be easier to read